pkg/database/database.go

```diff
@@ -103,6 +106,132 @@ func (db *Database) KeyManager() keys.KeyManager {
 	return db.keyManager
 }
 
+// GetCookieHashAndEncryptionKeys gets the cookie hash and encryption keys. The
+// first 32 bytes are the encryption key and the remaining bytes are the HMAC
+// key.
+func (db *Database) GetCookieHashAndEncryptionKeys(fallback []envconfig.Base64Bytes) ([][]byte, error) {
+	ctx, done := context.WithTimeout(context.Background(), 5*time.Second)
+	defer done()
+
+	results, err := db.secretResolver.Resolve(ctx, db, db.secretManager, SecretTypeCookieKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(sethvargo): remove in 0.28.0+
+	if len(results) == 0 {
+		existing := envconfigBytesToBytes(fallback)
+		if l := len(existing); l%2 != 0 {
+			return nil, fmt.Errorf("fallback bytes are not even (got %d)", l)
+		}
+
+		// In v1 cookies, the HMAC key is first, then the encryption key.
+		final := make([][]byte, len(existing)/2)
```

Comment:
I hate this.

Reply:
it gets removed in the next release at least :)